### PR TITLE
Staking xIon, Fix Testnet, bump protocol and version

### PR DESCRIPTION
**Pull request by cevap**
protocol versino updated, client version updated, xIon staking applied

more info about source patch: https://github.com/PIVX-Project/PIVX/commit/3b778f56fe2b5e5b363c8c6d794a848a84d630d1

orginal patch: [source-patch-for-check-0094-Staking-zPiv.txt](https://github.com/cevap/ion/files/2011563/source-patch-for-check-https://github.com/PIVX-Project/PIVX/commit/3b778f56fe2b5e5b363c8c6d794a848a84d630d10094-Staking-zPiv.txt)

This version compiles and syncs with and without bootstrap file.

**TODO**
add messages to different checks which are now empty/marked out
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312426%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312438%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312494%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312515%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312529%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312553%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312581%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312909%22%2C%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191313166%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20Makefile.am%2036%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312529%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20be%20%60ion.icns%60%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A13%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile.am%3AL86-93%22%7D%2C%20%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20Makefile.am%2020%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312494%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20be%20%60ion.icns%60%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A12%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile.am%3AL30-43%22%7D%2C%20%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20Makefile.am%2027%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312515%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20be%20%60ion.ico%60%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A13%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile.am%3AL30-43%22%7D%2C%20%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20Makefile.am%2011%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312438%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Mistakenly%20removed%2C%20should%20be%20added%20back.%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A12%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile.am%3AL1-10%22%7D%2C%20%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20Makefile.am%2054%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312581%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Mistakenly%20removed%2C%20should%20be%20added%20back.%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A13%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile.am%3AL225-230%22%7D%2C%20%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20Makefile.am%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312553%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20be%20%60ion.icns%60%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A13%3A25Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile.am%3AL99-106%22%7D%2C%20%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20Makefile.am%203%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312426%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Mistakenly%20removed%2C%20should%20be%20added%20back.%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A12%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%2C%20%7B%22body%22%3A%20%22Mistakenly%20removed%2C%20should%20be%20added%20back.%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A17%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile.am%3AL1-10%22%7D%2C%20%22Pull%202a21754ac6c3110bb07386adf289efc310f73d1e%20src/libzerocoin/CoinSpend.h%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/cevap/ion/pull/37%23discussion_r191312909%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Minor%3A%20this%20is%20duplicated%20and%20can%20be%20removed.%22%2C%20%22created_at%22%3A%20%222018-05-29T06%3A15%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/8009243%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/mitchellcash%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/libzerocoin/CoinSpend.h%3AL21-32%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e Makefile.am 3'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312426'>File: Makefile.am:L1-10</a></b>
- <a href='https://github.com/mitchellcash'></a> Mistakenly removed, should be added back.
- <a href='https://github.com/mitchellcash'></a> Mistakenly removed, should be added back.
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e Makefile.am 11'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312438'>File: Makefile.am:L1-10</a></b>
- <a href='https://github.com/mitchellcash'></a> Mistakenly removed, should be added back.
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e Makefile.am 20'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312494'>File: Makefile.am:L30-43</a></b>
- <a href='https://github.com/mitchellcash'></a> Should be `ion.icns`
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e Makefile.am 27'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312515'>File: Makefile.am:L30-43</a></b>
- <a href='https://github.com/mitchellcash'></a> Should be `ion.ico`
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e Makefile.am 36'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312529'>File: Makefile.am:L86-93</a></b>
- <a href='https://github.com/mitchellcash'></a> Should be `ion.icns`
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e Makefile.am 45'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312553'>File: Makefile.am:L99-106</a></b>
- <a href='https://github.com/mitchellcash'></a> Should be `ion.icns`
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e Makefile.am 54'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312581'>File: Makefile.am:L225-230</a></b>
- <a href='https://github.com/mitchellcash'></a> Mistakenly removed, should be added back.
- [ ] <a href='#crh-comment-Pull 2a21754ac6c3110bb07386adf289efc310f73d1e src/libzerocoin/CoinSpend.h 8'></a> &nbsp;<b><a href='https://github.com/cevap/ion/pull/37#discussion_r191312909'>File: src/libzerocoin/CoinSpend.h:L21-32</a></b>
- <a href='https://github.com/mitchellcash'></a> Minor: this is duplicated and can be removed.

<a href='https://www.codereviewhub.com/cevap/ion/pull/37?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/cevap/ion/pull/37?approve=1'></a>&nbsp;<a href='https://github.com/cevap/ion/pull/37'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by cevap** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/cevap/ion/pull/37'>Refresh</a>.<a href='#crh-update-none'></a>

File: Makefile.am [hunk 1]
```diff
@@ -1,14 +1,9 @@
 # Copyright (c) 2013-2016 The Bitcoin Core developers
-# Copyright (c) 2015-2018 The PIVX developers
-# Copyright (c) 2018 The Ion developers
```
Comment (mitchellcash) [inline Makefile.am]: Mistakenly removed, should be added back.

File: Makefile.am [hunk 2]
```diff
@@ -1,14 +1,9 @@
...
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 ACLOCAL_AMFLAGS = -I build-aux/m4
 SUBDIRS = src
-if ENABLE_MAN
-SUBDIRS += doc/man
-endif
```
Comment (mitchellcash) [inline Makefile.am]: Mistakenly removed, should be added back.

File: Makefile.am [hunk 3]
```diff
@@ -35,13 +30,13 @@ OSX_BACKGROUND_IMAGE_DPIS=36 72
 OSX_DSSTORE_GEN=$(top_srcdir)/contrib/macdeploy/custom_dsstore.py
 OSX_DEPLOY_SCRIPT=$(top_srcdir)/contrib/macdeploy/macdeployqtplus
 OSX_FANCY_PLIST=$(top_srcdir)/contrib/macdeploy/fancy.plist
-OSX_INSTALLER_ICONS=$(top_srcdir)/src/qt/res/icons/ion.icns
+OSX_INSTALLER_ICONS=$(top_srcdir)/src/qt/res/icons/bitcoin.icns
```
Comment (mitchellcash) [inline Makefile.am]: Should be `ion.icns`

File: Makefile.am [hunk 4]
```diff
@@ -35,13 +30,13 @@ OSX_BACKGROUND_IMAGE_DPIS=36 72
...
 OSX_PLIST=$(top_builddir)/share/qt/Info.plist #not installed
 OSX_QT_TRANSLATIONS = da,de,es,hu,ru,uk,zh_CN,zh_TW
 
 DIST_DOCS = $(wildcard doc/*.md) $(wildcard doc/release-notes/*.md)
 
-WINDOWS_PACKAGING = $(top_srcdir)/share/pixmaps/ion.ico \
+WINDOWS_PACKAGING = $(top_srcdir)/share/pixmaps/bitcoin.ico \
```
Comment (mitchellcash) [inline Makefile.am]: Should be `ion.ico`

File: Makefile.am [hunk 5]
```diff
@@ -91,7 +86,7 @@ $(OSX_APP)/Contents/Info.plist: $(OSX_PLIST)
 	$(MKDIR_P) $(@D)
 	$(INSTALL_DATA) $< $@
 
-$(OSX_APP)/Contents/Resources/ion.icns: $(OSX_INSTALLER_ICONS)
+$(OSX_APP)/Contents/Resources/bitcoin.icns: $(OSX_INSTALLER_ICONS)
```
Comment (mitchellcash) [inline Makefile.am]: Should be `ion.icns`

File: Makefile.am [hunk 6]
```diff
@@ -104,7 +99,7 @@ $(OSX_APP)/Contents/Resources/Base.lproj/InfoPlist.strings:
 	echo '{	CFBundleDisplayName = "$(PACKAGE_NAME)"; CFBundleName = "$(PACKAGE_NAME)"; }' > $@
 
 OSX_APP_BUILT=$(OSX_APP)/Contents/PkgInfo $(OSX_APP)/Contents/Resources/empty.lproj \
-  $(OSX_APP)/Contents/Resources/ion.icns $(OSX_APP)/Contents/Info.plist \
+  $(OSX_APP)/Contents/Resources/bitcoin.icns $(OSX_APP)/Contents/Info.plist \
```
Comment (mitchellcash) [inline Makefile.am]: Should be `ion.icns`

File: Makefile.am [hunk 7]
```diff
@@ -230,7 +225,5 @@ CLEANFILES = $(OSX_DMG) $(BITCOIN_WIN_INSTALLER)
 
 .INTERMEDIATE: $(COVERAGE_INFO)
 
-DISTCHECK_CONFIGURE_FLAGS = --enable-man
-
```
Comment (mitchellcash) [inline Makefile.am]: Mistakenly removed, should be added back.

File: src/libzerocoin/CoinSpend.h [hunk 8]
```diff
@@ -21,8 +21,11 @@
 #include "Commitment.h"
 #include "Params.h"
 #include "SerialNumberSignatureOfKnowledge.h"
+#include "SpendType.h"
 #include "bignum.h"
+#include "pubkey.h"
 #include "serialize.h"
+#include "SpendType.h"
```
Comment (mitchellcash) [inline src/libzerocoin/CoinSpend.h]: Minor: this is duplicated and can be removed.

File: Makefile.am (hunk 1 above)
Comment (mitchellcash) [inline Makefile.am]: Mistakenly removed, should be added back.

**Pull request closed without merging**